### PR TITLE
UI: Reset Glyph

### PR DESCRIPTION
**Pull request by tfamula**
Introducing a new Reset Glyph, coming from [Responsive Table Filters](https://www.ilias.de/docu/goto.php?target=wiki_1357_Responsive_Table_Filters#ilPageTocA121)

### CODE REVIEW

File: src/UI/Component/Glyph/Factory.php [hunk 1]
```diff
@@ -699,4 +699,38 @@ public function angry($action = null);
...
+	 *       Upon clicking the related control is reloaded immediately and goes back to state
+	 *       before the user has started to make changes.
+	 *
+	 * featurewiki:
+	 *       - https://www.ilias.de/docu/goto.php?target=wiki_1357_Responsive_Table_Filters#ilPageTocA121
+	 *
+	 * rules:
+	 *   usage:
+	 *       1: >
+	 *          The Reset Glyph SHOULD not come without an Apply Glyph and vice versa.
```
Comment (Amstutz) [inline src/UI/Component/Glyph/Factory.php:774]: Why?

File: src/UI/Component/Glyph/Factory.php [hunk 2]
```diff
@@ -699,4 +699,38 @@ public function angry($action = null);
...
+	 *       2: >
+	 *          If there are no changes to reset, the Reset Glyph MUST be deactivated (or not clickable).
+	 *   style:
+	 *       1: >
+	 *          The deactivated state of the Reset Glyph MUST be visually noticeable for the user, i.e. by
+	 *          greying out the Reset Glyph.
```
Comment (Amstutz) [inline src/UI/Component/Glyph/Factory.php:780]: Add aria state.

**Review by Amstutz — CHANGES_REQUESTED** [review]

**Comment by Amstutz** [comment]
Exactly the same reasoning as with the apply. Not sure about the choice of glyph the map the function, however since Christophe and Yvonne seem to be fine, I will not object.

Two things:
- Why does this has to always come with the reset glyph?
- Add some aria attribute for the disabled state (e.g. aria-disabled).

File: src/UI/Component/Glyph/Factory.php (hunk 1 above)
Comment (tfamula) [reply src/UI/Component/Glyph/Factory.php:774]: see [Apply Glyph](https://github.com/ILIAS-eLearning/ILIAS/pull/1122)

**Comment by matthiaskunkel** [comment]
We highly appreciate this suggestion and accept it for trunk. Please add the aria state to support accessiblility and introduce the 'withUnavailableAction' for glyphs before committing to trunk.

File: src/UI/templates/default/Glyph/glyph.less [hunk 3]
```diff
@@ -8,6 +8,12 @@
 		}
 	}
 
+	&.disabled {
+		opacity: 0.2;
+		filter: alpha(opacity=20); /* For IE8 and earlier */
```
Comment (alex40724) [inline src/UI/templates/default/Glyph/glyph.less]: No need to support IE8.

I guess a color: @il-glyph-disabled-color; would make more sense than using opacity.

@Amstutz What do you think?

File: src/UI/templates/default/Glyph/tpl.glyph.standard.html [hunk 4]
```diff
@@ -1,3 +1,3 @@
-<a class="glyph<!-- BEGIN highlighted --> highlighted<!-- END highlighted -->"<!-- BEGIN with_action --> href="{ACTION}"<!-- END with_action --> aria-label="{LABEL}"<!-- BEGIN with_id --> id="{ID}"<!-- END with_id -->>
+<a class="glyph<!-- BEGIN highlighted --> highlighted<!-- END highlighted --><!-- BEGIN disabled --> disabled<!-- END disabled -->"<!-- BEGIN with_action --> href="{ACTION}"<!-- END with_action --> aria-label="{LABEL}"<!-- BEGIN with_aria_disabled --> aria-disabled="{ARIA_DISABLED}"<!-- END with_aria_disabled --><!-- BEGIN with_id --> id="{ID}"<!-- END with_id -->>
```
Comment (alex40724) [inline src/UI/templates/default/Glyph/tpl.glyph.standard.html]: I think there is nothing like a disabled a-tag. We should use a span or omit the container tag completely instead imo.

Comment (alex40724) [reply src/UI/templates/default/Glyph/tpl.glyph.standard.html]: Ok, I saw the discussion in the other PR. We might keep it this way.

**Comment by Amstutz** [comment]
Hi @tfamula, I think this is ready to be merged, right? Please run the tests one final time and resolve the 3 conflicts and then we merge this. Thanks a lot for this contribution!

**Comment by tfamula** [comment]
Ready for merging, @alex40724 will do this and resolve the conflicts.

**Review by Amstutz — APPROVED** [review]

**Pull request merged**